### PR TITLE
feat: [CDS-53660]: integrated Nexus grupid and artifactid with apis and also set it behind feature flag

### PR DESCRIPTION
**Pull request by sahithikolichala**
…indff

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx [hunk 1]
```diff
@@ -331,7 +469,192 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
     </div>
   ))
 
+  const groupItemRenderer = memoize((item: { label: string }, { handleClick }) => (
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:472]: Use the common component here also that use in Nexus2

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx [hunk 2]
```diff
@@ -331,7 +469,192 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
...
+          </Layout.Horizontal>
+        }
+        disabled={fetchingGroupIds}
+        onClick={handleClick}
+      />
+    </div>
+  ))
+
+  const artifactIdItemRenderer = memoize((item: { label: string }, { handleClick }) => (
+    <div key={item.label.toString()}>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:487]: Same as above

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/NexusArtifactSource/NexusArtifactSource.tsx [hunk 3]
```diff
@@ -377,7 +517,192 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
     </div>
   ))
 
+  const groupItemRenderer = memoize((item: { label: string }, { handleClick }) => (
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/NexusArtifactSource/NexusArtifactSource.tsx:520]: Same as above

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/NexusArtifactSource/NexusArtifactSource.tsx [hunk 4]
```diff
@@ -377,7 +517,192 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
...
+            <Text>{item.label}</Text>
+          </Layout.Horizontal>
+        }
+        disabled={fetchingGroupIds}
+        onClick={handleClick}
+      />
+    </div>
+  ))
+
+  const artifactIdItemRenderer = memoize((item: { label: string }, { handleClick }) => (
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/NexusArtifactSource/NexusArtifactSource.tsx:534]: Same as above

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus3Artifact/Nexus3Artifact.tsx [hunk 5]
```diff
@@ -235,6 +292,56 @@ export function Nexus3Artifact({
...
+  useEffect(() => {
+    const artifactOptions: SelectOption[] = (artifactIdData?.data || [])?.map(artifact => {
+      return {
+        label: artifact,
+        value: artifact
+      } as SelectOption
+    }) || [
+      {
+        label: getString('common.loadingFieldOptions', {
+          fieldName: getString('common.subscriptions.tabs.plans')
```
[the remark this answers is not in the document]
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus3Artifact/Nexus3Artifact.tsx:335]: oops somehow these changes got merged, will fix this

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/__test__/Nexus2Artifact.test.tsx [hunk 6]
```diff
@@ -358,21 +358,8 @@ describe('Nexus Artifact tests', () => {
     const submitBtn = container.querySelector('button[type="submit"]')!
     fireEvent.click(submitBtn)
 
-    const queryByNameAttribute = (name: string): HTMLElement | null => queryByAttribute('name', container, name)
```
[the remark this answers is not in the document]
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/__test__/Nexus2Artifact.test.tsx:361]: i dont need to modify the test values are already updated

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 7]
```diff
@@ -951,6 +951,8 @@ selectArtifactPathPlaceholder: Select or Enter an Artifact Path (Optional)
 artifactPathFilterLabel: Artifact Path Filter
 artifactsSelection:
   loadingRepository: Loading Repository...
+  loadingGroupIds: Loading GroupIds...
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:954]: Check if we can delete this now

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/helper.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+interface RequestOptions {
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/helper.ts:8]: Check if we can use this from service index file.

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx [hunk 9]
```diff
@@ -120,6 +127,48 @@ export function NexusArtifact({
     }
   })
 
+  const {
+    data: groupIdData,
+    loading: fetchingGroupIds,
+    error: groupIdError,
+    refetch: refetchGroupIds
+  } = useMutateAsGet(useGetGroupIds, {
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx:135]: Use the helper here also

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx [hunk 10]
```diff
@@ -120,6 +127,48 @@ export function NexusArtifact({
...
+    lazy: true,
+    debounce: 300
+  })
+
+  const {
+    data: artifactIdData,
+    loading: fetchingArtifactIds,
+    error: artifactIdError,
+    refetch: refetchArtifactIds
+  } = useMutateAsGet(useArtifactIds, {
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx:156]: Same as above

**Review by pankajsingh-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by satyam-harness — DISMISSED** [review]

**Comment by sahithikolichala** [comment]
retrigger jest

**Review by YounixIO — APPROVED** [review]

**Pull request merged**